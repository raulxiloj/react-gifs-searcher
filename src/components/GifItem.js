import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {

    return (
        <div className="col-md-3 mb-4 align-items-stretch animate__animated animate__fadeIn">
            <div className="card" >

                <img className="card-img-top" src={url} alt={title} />

                <div className="card-body">
                    <p className="card-text">{title}</p>
                </div>

            </div>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}