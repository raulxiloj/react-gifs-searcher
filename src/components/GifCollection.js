import React from 'react'
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifCollection = ({ category }) => {//Gif grid

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    images.map(img => <GifItem key={img.id} {...img} />)
                }
            </div>
        </>
    )
}

GifCollection.propTypes = {
    category: PropTypes.string.isRequired
}