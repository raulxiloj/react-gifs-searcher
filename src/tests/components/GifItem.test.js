import React from 'react';
import { shallow } from 'enzyme'
import { GifItem } from '../../components/GifItem';


describe('Tests in GifCollection', () => {

    const title = 'some title';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifItem title={title} url={url} />);

    test('Should show the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('The parragraph should have the title text', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('The image should have the title and the url of the props', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    })

    test('should have the css class card', () => {
        const div = wrapper.find('div');
        expect(div.props().className.includes('card')).toBe(true);
    })

})