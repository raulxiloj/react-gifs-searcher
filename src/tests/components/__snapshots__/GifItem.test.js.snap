// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifCollection Should show the component correctly 1`] = `
<div
  className="card"
>
  <img
    alt="some title"
    src="https://localhost/algo.jpg"
  />
  <p>
    some title
  </p>
</div>
`;
